cpp/examples/parquet/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@
 add_executable(parquet-low-level-example low-level-api/reader-writer.cc)
 add_executable(parquet-low-level-example2 low-level-api/reader-writer2.cc)
 add_executable(parquet-arrow-example parquet-arrow/reader-writer.cc)
+add_executable(parquet-stream-api-example parquet-stream-api/stream-reader-writer.cc)
```

Comment:
this is potentially useful, but can you also please provide proper unit tests for this PR

Reply:
I added unit tests for the StreamWriter class.
For the StreamReader class I have not figured out where the Parquet test data files are located.  Can you help with this?

Reply:
I think since this is a wrapper, it is probably okay to generate test files in a temporary directory on the fly (see arrow read/writer tests) for examples.

Reply:
Ok I have added unit tests for both the StreamReader and StreamWriter class.